tutorial/chapter1.md

```diff
@@ -0,0 +1,200 @@
+    }  
+}
+```
+In this example there are some features of Klepsydra that will be introduced and others that have been described in the [kpsr-tutorial - chapter 1](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/master/tutorials/chapter1.md).
+
+* From the module kpsr-robotics/vision_ocv/modules/mem_mdlw, the ```kpsr::vision_ocv::ImageMultiplexerVisionMiddlewareProvider``` class is introduced. Note that a param of this class is ```kpsr::high_performance::DataMultiplexerMiddlewareProvider```, which is a single-producer, multiple-consumer introduced in the [kpsr-tutorial - chapter 1](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/master/tutorials/chapter1.md).
+
+* From the module kpsr-robotics/vision_ocv/modules/vision_ocv, ```kpsr::vision_ocv::ImageDataFactory```class is introduced. It is used to handle the allocation of images, copy and resizing.
+
+* The ```kpsr::Service``` was introduced in the [kpsr-tutorial - chapter 1](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/master/tutorials/chapter1.md) and it is used in this example (by means of inheritance) to create two custom services:
+
+   * ```FileImageStreamingService```, which is a service that reads images from a folder and it publishes them
+   * The five ```kpsr_tutorial::ImageProcessingService```. Each one of these services has the purpose of simulating a image processing algorithm . So, in this example, each service will process images and  will calculate the pixel mean of these images (in this example these images are read and publish by FileImageStreamingService). After that, each service will publish a data which contains this mean. 
+   
+* Note that there are five ```kpsr_tutorial::ImageProcessingService``` to show the behavioral of the example in means of threads and Data Multiplexer features. So:
+   * each subscriber of this service (Data Multiplexer Consumers) is handled by one thread (so, if there are five consumers, it will be five threads associated to these consumers).
+   * the second argument of the constructor of these services represents the delay (in milliseconds) added to the task that these services do. So, the higher delay, the longer the thread will take to process the image. Due to the Data Multiplexer behavioral, the service with higher delay will process less images (because it will spend more time processing and it will discard the images it can't receive). In this example the Image Processing Service 1 will receive more images than any other service (it has the lower delay: 50 [ms]), whereas the Image Processing Service 2 will receive less images than any other service (it has the higher delay: 10000 [ms]).
+
+* Finally, it is important to analyze the number of threads running in this example to understand the behavioral of the Klepsydra Vision API. So, there are nine threads running in this example:
```

Comment:
Finally, we can analyze the number of threads running in this example to further understand the behaviour ..